lib/runtime/wasmer/imports.go

```diff
@@ -185,45 +188,191 @@ func ext_sandbox_memory_teardown_version_1(context unsafe.Pointer, a C.int32_t)
 }
 
 //export ext_crypto_ed25519_generate_version_1
-func ext_crypto_ed25519_generate_version_1(context unsafe.Pointer, a C.int32_t, z C.int64_t) C.int32_t {
+func ext_crypto_ed25519_generate_version_1(context unsafe.Pointer, keyTypeID C.int32_t, seedSpan C.int64_t) C.int32_t {
 	logger.Trace("[ext_crypto_ed25519_generate_version_1] executing...")
-	logger.Warn("[ext_crypto_ed25519_generate_version_1] unimplemented")
-	return 0
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	runtimeCtx := instanceContext.Data().(*runtime.Context)
+
+	// TODO: key types not yet implemented
+	// id := memory[idData:idData+4]
```

Comment:
key types are now implemented (if you look in lib/keystore/keystore.go the `Name` field is the 4-byte ID) however it'll require changing the keystore that's passed to the runtime to be the `GlobalKeystore` for the node, instead of just the `Acco` keystore that's currently passed. I'll open an issue for this and it can be done separately